linux/aux-tools/startup-misc.sh

```diff
@@ -17,3 +17,32 @@ fi
 
 # Hide mounted devices from qubes-block list (at first udev run, only / is mounted)
 udevadm trigger --action=change --subsystem-match=block
+
+if [ -n "$(qvm-features dom0 suspend-s0ix)" ]; then
+    # assign thunderbolt root ports to pciback as workaround for suspend
+    # issue without PCI hotplut enabled, see
+    # https://github.com/QubesOS/qubes-linux-kernel/pull/903 for details
+    for dev in /sys/bus/pci/devices/0000:00:*; do
+        [ -h "$dev" ] || continue
+        sbdf=$(basename "$dev")
+        # PCIe bridge
+        if [ "$(cat "$dev/class")" = "0x060400" ]; then
+            # There seems to be no property saying it's thunderbolt other than product id...
+            lspci -s "$sbdf" | grep -q Thunderbolt || continue
+
+            if [ -e "$dev/driver" ]; then
+                echo "$sbdf" > "$dev/driver/unbind"
+            fi
+            echo "$sbdf" > /sys/bus/pci/drivers/pciback/new_slot
+            echo "$sbdf" > /sys/bus/pci/drivers/pciback/bind
+            echo "$sbdf" > /sys/bus/pci/drivers/pciback/qubes_exp_pm_suspend
+            echo "$sbdf" > /sys/bus/pci/drivers/pciback/qubes_exp_pm_suspend_force
+        elif [ "$(cat "$dev/class")" = "0x0c0340" ]; then
+            # Thunderbolt 4 NIH device
+            # it's already assigned to pciback by initramfs, so just set the
+            # flag (unless user deassigned it explicitly)
+            [ "$(basename "$(readlink "$dev/driver")")" = "pciback" ] || continue
```

Comment:
This was false assumption, the device got assigned to the `thunderbolt` driver.

Reply:
That’s not good!

Reply:
Oh, that's actually expected, because I (intentionally) do _not_ have `qubes.hide_all_usb` option set there. So, I need to adjust the script to handle that condition too.

Reply:
Phew!!!!  I was scared this was a security vulnerability in Qubes OS.